src/test/create-user-test.ts

```diff
@@ -0,0 +1,151 @@
+import axios, { AxiosResponse } from 'axios';
+import bcrypt from 'bcrypt';
+import { expect } from 'chai';
+import { serverUrl } from '../setup-server';
+import { AppDataSource } from '../data-source.js';
+import { User } from '../entity/User.js';
+
+interface InputData {
+  name: string;
+  email: string;
+  birthDate: string;
+  password: string;
+}
+
+async function postQuery(inputData: InputData): Promise<AxiosResponse> {
+  const response = await axios.post(`${serverUrl}graphql`, {
+    query: `
+      mutation CreateUser($data: UserInput!) {
+      createUser(data: $data) {
+          id
+          name
+          email
+          birthDate
+        }
+      }
+      `,
+    variables: {
+      data: inputData,
+    },
+  });
+
+  return response;
+}
+
+async function checksInputAndReturnedUser(inputData: InputData, response: AxiosResponse) {
+  const { id, ...createUser } = response.data.data.createUser;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { password, ...inputDataWithoutPassword } = inputData;
+
+  expect(createUser).to.deep.equal(inputDataWithoutPassword);
+  expect(Number(id)).to.be.above(0);
+}
+
+async function checksInputAndStoredUser(inputData: InputData) {
+  const userRepository = AppDataSource.getRepository(User);
+
+  const storedUser = await userRepository.findOneBy({
+    email: inputData.email,
+  });
+
+  const storedUserWithoutIdAndPassword = {
+    name: storedUser.name,
+    email: storedUser.email,
+    birthDate: storedUser.birthDate,
+  };
+
+  const inputDataWithoutPassword = {
+    name: inputData.name,
+    email: inputData.email,
+    birthDate: inputData.birthDate,
+  };
+
+  expect(storedUserWithoutIdAndPassword).to.deep.equal(inputDataWithoutPassword);
+  expect(Number(storedUser.id)).to.be.above(0);
+  expect(await bcrypt.compare(inputData.password, storedUser.password)).to.be.true;
+}
+
+describe('createUser mutation', () => {
+  before(async () => {
```

Comment:
aqui seria um beforeEach, não?
a ideia é apgar antes de cada `it`

Reply:
Na verdade antes até estava um beforeEach, mas não era o comportamento que eu esperava porque no `it` seguinte eu tento criar um usuário com o mesmo email e ele deveria dar um erro. Se eu limpo o banco antes dele a criação seria feita com sucesso e o teste não se comportaria da forma que eu desejava.